test/linalg/matmul.jl

```diff
@@ -389,3 +387,15 @@ let
         @test_throws DimensionMismatch A_mul_B!(full43, full43, tri44)
     end
 end
+
+# Ensure that matrix multiplication with a narrower output type than input type does not
+# produce allocation in the inner loop (#14722), by ensuring allocation does not change
```

Comment:
this got deleted from master because it was making coverage fail when inlining was off